q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,322 @@
+
+import yaml
+from glob import glob
+import pandas as pd
+from os.path import basename, isdir, splitext
+
+from q2_metadata.normalization._norm_messages import warningsCollection, errorsCollection
+
+
+class Rules(object):
+    """Collect the rules of one current variable rules.
+
+    Parameters
+    ----------
+    variable_rules_fp : str
+        Path to one variable's rules yaml file.
+
+    Attributes
+    ----------
+    variable : str
+        Name of the variable which rules are parsed and
+        that is the yaml file name without the .yml extension.
+
+    remap : dict
+        Apply the rules of remapping variables values, by replacing
+        all instances of a given value by another one value.
+                    'remap:
+                      str: str'
+    validation : dict
+        Apply the rules of validating variables values, by replacing by the blank
+        (or the missing) value every instance which in one or many other columns
+        have a null or missing value.
+                    'validation:
+                      force_to_blank_if:
+                        is null:
+                        - variable'
+    normalization : dict
+        Apply the rules of normalizing the variables values, by replacing
+        the values outside the range by a given replacement value.
+                    'normalization:
+                        gated_value:
+                            - Out of bounds
+                        maximum:
+                            - int
+                            - float
+                        minimum:
+                            - int
+                            - float'
+    missing : str
+        Apply the rules of checking the allowed missing values, by replacing
+        the not-expected values and numpy's nan (empty) by the passed
+        missing value term.
+                    'missing:
+                        - not applicable
+                        - not collected
+                        - not provided
+                        - restricted access'
+    format : str
+        Apply the rules of checking that the passed column in the correct format.
+        [THIS COMMAND WOULD EITHER INFER THE DTYPE (as of now) OR USE THE
+        RESULT OF AN INITIAL DTYPE GETTER FUNCTION (see _dtypes.py)]
+                    'format:
+                      - bool
+                      - float
+                      - int
+                      - str'
+    expected : list
+        Apply the rules of allowing only expected values.
+        These do allow the missing and blanks.
+                    'expected:
+                        - str
+                        - list'
+    ontology : str
+        Name of the ontology to lookup within for control vocabulary, e.g.
+                    'ontology: Gazetteer'
+    blank : str
+        Apply the rules of checking the allowed blank values, by replacing
+        the not-expected values and numpy's nan (empty) by the passed
```

Comment:
how is blank different than missing?

Reply:
to my understanding, blank is the allowed labelling for the DNA extraction or sequencing  controls samples, while missing could be any other failure reason (e.g. height not measure because the ruler was broken) - does that make sense?

also - this was not an attribute - it is now indented as a docstring intro text.

Reply:
we need to find out whether there should be a control vocabulary for the blanks, e.g. metabolomics people use "SOLVENT" so it might well be free text. Yet, identifying blanks in a curation process is critical as we may not apply all rules similarly to blanks vs. assays.